src/textual/widgets/_option_list.py

```diff
@@ -647,6 +655,51 @@ def remove_option_at_index(self, index: int) -> Self:
             ) from None
         return self
 
+    def _replace_option_prompt(self, index: int, prompt: RenderableType) -> None:
+        """Replace the prompt of an option in the list.
+
+        Args:
+            index: The index of the option to replace the prompt of.
+            prompt: The new prompt for the option.
+
+        Raises:
+            IndexError: If there is no option of the given index.
```

Comment:
I believe this is incorrect, right? What `self.get_option_at_index` might raise is `OptionDoesNotExist`.

Reply:
Nice catch! Thank you for pointing that out. It's fixed now.